src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -116,7 +231,28 @@ public void printGradesForGroup(String groupCode) {
      * 179234 Dawid Donald
      * 189521 Anna Kowalska
      */
-    public void printAllStudents() {
 
+    public void printAllStudents() {
+        Student[] tmp = new Student[studentsPrintListCounter];
+        for (int i = 0; i < studentsPrintListCounter; i++) {
+            for (int j = studentsPrintListCounter - 1; j > 0; j--) {
+                if (studentsPrintList[j].getIndex() < studentsPrintList[j - 1].getIndex()) {
+                    tmp[0] = studentsPrintList[j];
+                    studentsPrintList[j] = studentsPrintList [j - 1];
+                    studentsPrintList [j - 1] = tmp[0];
+                }
+            }
+        }
+        for (int i = 0; i < studentsPrintListCounter - 1; i++) {
+            if (studentsPrintList[i].getIndex() == studentsPrintList[(i + 1 % studentsPrintList.length)].getIndex()) {
+                studentsPrintList[i] = null;
+                i++;
+            }
+        }
+        for (int i = 0; i < studentsPrintListCounter; i++) {
+            if (studentsPrintList[i] !=  null) {
+                System.out.println(studentsPrintList[i].studentInfo());
+            }
+        }
```

Comment:
Z tego co. rozumiem, to tutaj najpierw sortujesz tablicę i odsiewasz z niej duplikaty. Raczej powinieneś zadbać o to, żeby nie dodawać dwa razy studenta do tablicy `students`. Student może być w wielu grupach, ale tutaj powinien występować tylko raz
Czyli w metodzie `addStudentToGroup()` musiałbyś mieć logikę typu:
```
Student student = findStudent();
if (student == null) {
  ...dodaj studenta na kolejne miejsce w tablicy
}
```
powyższe oczywiście najlepiej wydzielić do osobnej metody typu `private void addStudent(Student student)`